pkg/updater/repository/repository.go

```diff
@@ -36,6 +48,14 @@ const (
 // is cleaned up during the next operation.
 type Repository struct {
 	RootPath string
+	m        sync.Mutex
```

Comment:
Let's remove this mutex and lock the GC call in the `updater`. That's the current pattern and will reduce the risk for deadlocks